app/frontend/src/features/profile/view/useReferences.ts

```diff
@@ -0,0 +1,54 @@
+import { Error as GrpcError } from "grpc-web";
+import { useQuery } from "react-query";
+
+import { Reference, User } from "../../../pb/api_pb";
+import { service } from "../../../service";
+import { timestamp2Date } from "../../../utils/date";
+import useUsers from "../../userQueries/useUsers";
+import { referencesQueryStaleTime } from "../constants";
+
+export default function useReferences(user: User.AsObject) {
+  const referencesQuery = useQuery<Reference.AsObject[], GrpcError>({
+    queryKey: ["references", { userId: user.userId }],
+    queryFn: async () => {
+      const referenceResponses = await Promise.all([
+        service.user.getReferencesReceived({
+          count: 50,
+          userId: user.userId,
+          offset: 0,
+        }),
+        service.user.getReferencesGiven({
+          count: 50,
+          userId: user.userId,
+          offset: 0,
+        }),
```

Comment:
I thought for now, rather than having "all" needing to combine the two reference type lists (given/received), then have "given" and "received" as two separate queries (so the filter becomes a thing to control which query in the cache to flick between), it's easier to combine them and do the filtering on the frontend.

Not feeling 100% about this though, the main limitation being doing pagination with this might be a pain... but maybe not with `useInfiniteQuery` eventually as the combined results themselves can be its own page?